Create ui-c8y-1022-46-0-validate-password-reset-token-before-showing-change-password-page.md (#3863)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1022-46-0-validate-password-reset-token-before-showing-change-password-page.md

---
date: ""

> title: Validate password reset token for more suitable redirection

Reviewer: ```suggestion
title: Password reset links now validated for a more suitable redirection
```

> title: Validate password reset token before showing change password page

Reviewer 2: ```suggestion
title: Validate password reset token for more suitable redirection
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63950
version: 1022.46.0
---

> There is new backend feature that allows Login app to check if reset password link is valid. If it is, user is redirected to view where he's able to change password. If it is invalid (does not exist or expired), user is instructed to request reset password link again.

Reviewer 2: ```suggestion
Previously, if user used password reset link with invalid token, he was redirected to reset password view and got error message on submit. To enhance user experience, the Login app now validates password reset links before allowing users to change their passwords. When a user clicks on a password reset link, the app checks with the backend to ensure the link is valid and has not expired. If the link is valid, the user is redirected to a page where they can set a new password. However, if the link is invalid or has expired, the user is redirected to different view and instructed to request a new password reset link.
```

> Previously, if a user used a password reset link with an invalid token, he was redirected to the reset password view and got an error message on submit. To enhance the user experience, the login app now validates password reset links before allowing users to change their passwords. If the link is valid and has not expired, the user is redirected to the reset password view to set a new password. If the link is invalid or has expired, the user is instructed to request a new password reset link.

Reviewer 2: ```suggestion
Previously, if a user used a password reset link with an invalid token, he was redirected to the reset password view and got an error message on submit. To enhance the user experience, the login app now validates password reset links before allowing users to change their passwords. If the link token is valid and has not expired, the user is redirected to the reset password view to set a new password. If the link token is invalid or has expired, the user is instructed to request a new password reset link.
```

> Previously, if a user used a password reset link with an invalid token, he was redirected to the reset password view and got an error message on submit. To enhance the user experience, the login app now validates password reset links before allowing users to change their passwords. If the link token is valid and has not expired, the user is redirected to the reset password view to set a new password. If the link token is invalid or has expired, the user is instructed to request a new password reset link.

Reviewer: ```suggestion
Previously, if users used a password reset link with an invalid token, they were redirected to the reset password view and got an error message on submit. To enhance the user experience, the login app now validates password reset links before allowing users to change their passwords. If the link token is valid and has not expired, the user is redirected to the reset password view to set a new password. If the link token is invalid or has expired, the user is instructed to request a new password reset link.
```

> Previously, if user used password reset link with invalid token, he was redirected to reset password view and got error message on submit. To enhance user experience, the Login app now validates password reset links before allowing users to change their passwords. When a user clicks on a password reset link, the app checks with the backend to ensure the link is valid and has not expired. If the link is valid, the user is redirected to a page where they can set a new password. However, if the link is invalid or has expired, the user is redirected to different view and instructed to request a new password reset link.

Reviewer: ```suggestion
Previously, if a user used a password reset link with an invalid token, he was redirected to the reset password view and got an error message on submit. To enhance the user experience, the login app now validates password reset links before allowing users to change their passwords. If the link is valid and has not expired, the user is redirected to the reset password view to set a new password. If the link is invalid or has expired, the user is instructed to request a new password reset link.
```

Reviewer: @jdre-c8y Just realized, it should be "token" instead of link in the last two sentences, right? "If the token is valid/invalid"

Reviewer 2: I mean, there is a token in the link, e.g. reset password links look like:
https://test.latest.stage.c8y.io?email=test.test@gmail.com&token=109ce5ee08a51ff6ad8fcc276d585f43bfe937df091c39b062d1f56b0a94880c

and the token parameter is validated . so it can be 'link token' in last sentence
